### PR TITLE
ENH: NeXus writer

### PR DESCRIPTION
**Pull request by prjemian**
NeXus HDF5 writer, templated from hdf5.py, basic unit tests all pass with Py3.6

### CODE REVIEW

**Comment by prjemian** [comment]
* py2.7 failed : Is this supported?
* py3.4 passed
* py3.5 failed : conda create env ....
* py3.6 : not tested by travis-ci

**Comment by prjemian** [comment]
@tacaswell @danielballan 
This is ready for you to review (and comment or merge)

File: pete.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+'''
```
Comment (tacaswell) [inline pete.py]: This file probably should not end up in the repo

**Comment by prjemian** [comment]
also checked one of the HDF5 files: plots with PyMCA and NeXpy, NeXus structure validates with punx

**Comment by tacaswell** [comment]
We are starting to add 3.6 to all of our projects, still working through making sure all of our dependencies are built for 3.6.

We do intend to support 2.7 (at least until 2020) even though we do not use it internally, but if replacing `fullmatch` with a 2-compatible version is too onerous, just mark the test as 3 only (see https://github.com/NSLS-II/databroker/blob/master/databroker/tests/test_broker.py#L25)

attn @stuartcampbell for review of this as the BNL nexus expert.

comment by 👻 Tom (real Tom is on vacation today).

**Comment by stuartcampbell** [comment]
At first glance this format looks similar to something we were looking at using for a HDF5 backed eventstore.  I think we should drop our idea in favour of this @tacaswell @arkilic what do you think ?

**Comment by tacaswell** [comment]
Nominally 👍 (I have not dug into the details of the file layout).

I think being able to have a directory of these files sitting behind a `DataBroker` would be a good point of collaboration.

**Comment by prjemian** [comment]
bump

**Comment by prjemian** [comment]
bump - this PR is feeling very lonely

**Comment by danielballan** [comment]
@arkilic, do you still plan to take a look at this?

**Comment by arkilic** [comment]
I did give 👍 to @stuartcampbell in person. I am happy with @prjemian's implementation. This makes it easier for us to collaborate and we are adopting this for the eventstore backend. If you're also happy with it, hit the green button @danielballan

**Comment by stuartcampbell** [comment]
@prjemian the only thing I might say is that we might need minor changes on the format (but also we might not) but just a heads up.

**Comment by prjemian** [comment]
That's good.  Thanks.

**Pull request merged**